drivers/pds/pdsbackup/backuputils.go

```diff
@@ -0,0 +1,217 @@
+package pdsbackup
+
+import (
+	"net/url"
+	"os"
+	"time"
+
+	pds "github.com/portworx/pds-api-go-client/pds/v1alpha1"
+	pdsapi "github.com/portworx/torpedo/drivers/pds/api"
+	"github.com/portworx/torpedo/drivers/pds/pdsutils"
+	"github.com/portworx/torpedo/pkg/log"
+)
+
+const (
+	bucketName         = "pds-qa-automation"
+	awsS3endpoint      = "s3.amazonaws.com"
+	bkpTimeOut         = 30 * time.Minute
+	bkpTimeInterval    = 10 * time.Second
+	bkpMaxtimeInterval = 60 * time.Second
+)
+
+// BackupClient struct
+type BackupClient struct {
+	controlPlaneURL    string
+	components         *pdsapi.Components
+	awsStorageClient   *awsStorageClient
+	azureStorageClient *azureStorageClient
+	gcpStorageClient   *gcpStorageClient
+}
+
+// CreateAwsS3BackupCredsAndTarget create backup creds,bucket and target.
+func (backupClient *BackupClient) CreateAwsS3BackupCredsAndTarget(tenantId, name string) (*pds.ModelsBackupTarget, error) {
+	log.Info("Add AWS S3 backup credentials")
+	akid := backupClient.awsStorageClient.accessKey
+	skid := backupClient.awsStorageClient.secretKey
+	region := backupClient.awsStorageClient.region
+	backupCred, err := backupClient.components.BackupCredential.CreateS3BackupCredential(tenantId, name, akid, awsS3endpoint, skid)
+	log.Infof("%v created successfully.", backupCred.GetName())
+	if err != nil {
+		log.Errorf("Error in adding the backup credentials to PDS , Err: %v ", err)
```

Comment:
do not log error, use fmt.Errorf() while returning error

Reply:
done